src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzerBase.cs

```diff
@@ -329,6 +342,31 @@ void VerifyDocumentationSyntax(Action<Diagnostic> reportDiagnostic, SyntaxNode s
             }
         }
 
+        private static IEnumerable<INamespaceSymbol> GetContainingNamespaces(ISymbol symbol)
+        {
+            INamespaceSymbol? currentNamespace = symbol.ContainingNamespace;
+
+            while (currentNamespace is { IsGlobalNamespace: false })
+            {
+                foreach (var constituent in currentNamespace.ConstituentNamespaces)
+                    yield return constituent;
+
+                currentNamespace = currentNamespace.ContainingNamespace;
+            }
+        }
+
+        private static IEnumerable<ISymbol> ExpandConstituentNamespaces(ISymbol symbol)
```

Comment:
Consider moving this method inside `BanFileEntry` type.

Reply:
Done.